src/ensembl/src/shared/components/instant-download/instant-download-gene/InstantDownloadGene.tsx

```diff
@@ -117,8 +128,10 @@ const InstantDownloadGene = (props: Props) => {
 
     try {
       await fetchForGene(payload);
+      props.onDownloadSuccess?.(payload);
     } finally {
       resetCheckboxes();
+      props.onDownloadFailure?.(payload);
```

Comment:
the `finally` clause runs regardless of whether an error happened within `try`.

Reply:
Thanks for noticing this! Should've gone into `catch`.